examples/state-management/agent.ts

```diff
@@ -0,0 +1,42 @@
+import { GameAgent } from "@virtuals-protocol/game"
+import dotenv from "dotenv";
+dotenv.config({ path: __dirname + '/.env' });
+
+import { HeadChef, IngredientManager } from "./worker"
+
+let moves = 10;
+const getAgentState = async () => {
+    return {
+        moves_left: moves
+    }
+}
+
+export const setAgentState = (state: any) => {
+    moves = state.moves_left;
+}
+
+export const updateMoves = (amount: number) => {
+    moves = amount;
+}
+
+export const getMoves = () => {
+    return moves;
+}
+
+if (!process.env.API_KEY) {
+    throw new Error('API_KEY is required in environment variables');
+}
+
+export const agent = new GameAgent(process.env.API_KEY, {
+    name: "kitchen_manager",
+    description: `You are a responsible kitchen manager that maximizes the efficiency of the workers under you and the kitchen.
+    You are a great problem solver and also nice to work with. 
+    You are also in charge of making sure the kitchen produces as much high quality food in the moves_left constraint, 
+    keep in mind that the moves_left constraint is the number of moves you have left to make, and each call to a worker reduces the moves_left constraint by 1`,
```

Comment:
Hey @jxx016, this is interesting! Ran through your example but noticed that the agent keep on running even though its `moves_left` decreased passed 0. I'd suggest adding a negative point restriction for the agent, wdyt?<br/>
<img width="1417" alt="image" src="https://github.com/user-attachments/assets/70bdd098-9bbc-4ccc-869a-b86a57e293fc" />

```suggestion
    description: `You are a responsible kitchen manager that maximizes the efficiency of the workers under you and the kitchen.
    You are a great problem solver and also nice to work with. 
    You are also in charge of making sure the kitchen produces as much high quality food in the moves_left constraint, 
    keep in mind that the moves_left constraint is the number of moves you have left to make, and each call to a worker reduces the moves_left constraint by 1
    NO MATTER WHAT, you must make sure that the you STOP OPERATING when the moves_left <= 0. NO NEGATIVE MOVES_LEFT ALLOWED.
    If negative moves_left is detected, STOP PROCEEDING and STOP OPERATING.`,
```

Reply:
Hey hey thanks for looking @Ang-dot . Very weird cuz it stopped for me when i tested before push, but might be inconsistent, or i might have mess something up after context switching a few times haha

But yes your suggestion is valid! I will apply it and try and test it out again tomorrow! Will test it a few times to ensure consistent behavior too.